newIDE/app/src/SceneEditor/index.js

```diff
@@ -455,7 +474,21 @@ export default class SceneEditor extends React.Component<Props, State> {
   _addInstance = (pos: [number, number], objectName: string) => {
     if (!objectName || !this.editor) return;
 
-    this.editor.addInstances(pos, [objectName]);
+    const instances = this.editor.addInstances(pos, [objectName]);
+    instances.forEach(instance => {
+      const infoBarDetails = onInstanceAdded(
+        instance,
+        this.props.layout,
+        this.props.project
+      );
+      if (infoBarDetails) {
+        this.setState({
+          additionalWorkInfoBar: infoBarDetails,
+          showAdditionalWorkInfoBar: true,
+        });
+      }
+    });
```

Comment:
Can you replace this by `this._onInstancesAdded(instances)`?

Reply:
you would need to remove the saveToHistory and updateToolbar because it's already done then.